dask/base.py

```diff
@@ -269,6 +269,9 @@ def normalize_function(func):
         return function_cache[func]
     except KeyError:
         result = _normalize_function(func)
+        if len(function_cache) > 10000:  # clear half of cache if full
```

Comment:
This seems excessively large. Perhaps a smaller number (say 500?). If the point is to cache for hot loops, I don't think we need to keep around that many.